wrapper.py

```diff
@@ -78,72 +79,102 @@ def main(logdir: str, cleanup: bool):
+                rnaseq_samples[sample_uniqID]['state'] = 'running'
+
+            # Set existing bioinfo record to 'running' and keep track of it
+            elif attached_bioinfo_object.cntn_cstm_SecondaryAnalysisState.value == 'novel':
+                new_bioinfo_record = update_bioinformatics_record(attached_bioinfo_object, fields={'cntn_cstm_SecondaryAnalysisState': 'running'})
+                rnaseq_samples[sample_uniqID]['bioinformatics'] = new_bioinfo_record
+                rnaseq_samples[sample_uniqID]['state'] = 'running'
+
+            else:  # State != novel
+                continue
+
+            # Save all the relevant fastq paths for this sample
+            if len(rnaseq_samples[sample_uniqID]) > 0:
+                all_fastq_paths = find_fastq_paths(slimsinfo.fastqs)
+                rnaseq_samples[sample_uniqID]['fastq'] = all_fastq_paths
+
     if len(rnaseq_samples) > 0:
-        logger.info(f"Found {len(rnaseq_samples)} sample(s) marked for QD-RNAseq pipeline. Starting the wrapper.")
+        logger.info(f"Found {len(rnaseq_samples.keys())} sample(s) marked for QD-RNAseq pipeline. Starting the wrapper.")
     else:
         sys.exit(0)
 
-    ### --- Loop over each record --- ###
-    runner_samples = {} # Store all samplesheet paths per sample in a dict for later use
-    for sample, record in rnaseq_samples.items():
+
+    ### --- Loop over each record, gather information needed --- ###
+    for sample in rnaseq_samples:
         ### --- Collect information about the sample --- ###
-        # Create a new SlimsSample object
         logger.info(f"{sample} - Extracting SLIMS information.")
-        slimsinfo = SlimsSample(sample)
-
-        # Translate the information from the SLIMS database into a dictionary
-        info = translate_slims_info(record)
-
-        # Get the run tag and combine with name
-        runtag = find_runtag_from_fastqs(slimsinfo.fastqs)
-        sample_id = f"{sample}_{runtag}"
+        fastq_paths = rnaseq_samples[sample]['fastq']
 
         ### --- Generate a samplesheet from the information gathered --- ###
-        outdir = os.path.join(config.get("general", "output_dir"), sample_id)
-        logger.info(f"{sample} - Generating samplesheet.csv in {outdir}.")
+        outdir = os.path.join(config.get("general", "output_dir"), sample)
+        logger.info(f"{sample} - Generating samplesheet.csv in {outdir}")
         os.makedirs(outdir, exist_ok=True)
 
         strandedness = config.get("general", "strandedness")
 
-        sample_ss_path = make_samplesheet(sample_id, slimsinfo.fastqs, strandedness, outdir)
-        runner_samples[sample_id] = sample_ss_path
+        sample_ss_path = make_samplesheet(sample, fastq_paths, strandedness, outdir)
+        rnaseq_samples[sample]['samplesheet'] = sample_ss_path
+
+
+    # ### --- Start a runner for each sample --- ###
+    # completed_samples = start_runner_threads(runner_samples, logger)
```

Comment:
So the whole way that the wrapper interacts with the runner is to be reworked. The idea being that the wrapper should do less, so that when we want to trigger pipes more directly from SLIMS, the runner does the heavy lifting.

Will leave this for future PR